### PR TITLE
Merge GoogleHadoopSyncableOutputStream functionality in the GoogleHadoopOutputStream

### PR DESCRIPTION
**Pull request by medb**

### CODE REVIEW

**Comment by medb** [comment]
> `spark.history.fs.gs.outputstream.type=FLUSHABLE_COMPOSITE` is used for some benchmarks and as a work around for PHS today.
> 
> Is there an alternative for that, if we plan to deprecate the option ? cc : @abmodi @ranu010101

After this change this is an obsolete option because this functionality is available in a single output stream, so it will work as expected out of the box. We may still want to have PHS-specific sync interval, but I think that this is optional and we can add it later if necessary.

File: gcs/CHANGES.md [hunk 1]
```diff
@@ -64,6 +64,13 @@
     fs.gs.auth.refresh.token
     ```
 
+1.  Merge all output stream types functionality in the default output stream,
+    and remove obsolete property:
```
Comment (cyxxy) [inline gcs/CHANGES.md:68]: Please add which of the original 3 types is the new default now.

**Review by cyxxy — APPROVED** [review]

File: gcs/CHANGES.md (hunk 1 above)
Comment (medb) [reply gcs/CHANGES.md]: Done.

**Review by mprashanthsagar — APPROVED** [review]

**Pull request merged**